Makefile

```diff
@@ -24,7 +24,7 @@ sort-dictionary:
 	LC_ALL=C sort ${SORT_ARGS} -u -o ${DICTIONARY} ${DICTIONARY}
 
 trim-dictionary:
-	sed -E -i.bak -e 's/^[[:space:]]+//; s/[[:space:]]+$$//; /^$$/d' ${DICTIONARY} && rm ${DICTIONARY}.bak
```

Comment:
What's wrong with just sed -E -i -e ...?
https://linux.die.net/man/1/sed

Reply:
> What's wrong with just sed -E -i -e ...?

That won't work on MacOS.  Mac has a BSD-derived sed, while Linux distros use the GNU version derived from ATT.   There seems to be no portable way to do sed in-place editing without making backup files.  Mac users could install gnu sed, but that doesn't seem like a reasonable dependency.

The change in commit 2d83ab7 switches from sed to ex, which should work on both Linux and MacOS (and others).  It does in-place editing properly, not creating a backup file in the first place.

Reply:
Ah yes, now you mention it, I think I've hit this myself somewhere.

You think `ex` is preferable or more cross-platform to `perl -pie`? I don't know, but I suspect `perl -pie` won't fail on no hits. I'd be rather wary about running in quiet mode AND having `|| true` on the end, that still feels like a recipe for trouble.

Otherwise is $$ or $RANDOM cross-platform enough to use as the extension with sed?

Reply:
@peternewman `$$` and `$RANDOM` are definitely cross-platform enough.  I think they would be fine in practice, giving a very small chance of clobbering an existing backup file, but as long as I was going to fix the issue, I thought I would do it right, i.e. by not specifying one at all, completely eliminating the risk. This also parallels the behavior of the `sort-dictionary` target.

I don't know Perl well enough to say whether `perl -pie` is preferable to `ex`, or which versions of `perl` are installed by default on which versions of different OSes.  The Perl substitution command would be simple enough, but (as usual) it is the error handling and rare conditions that are the tricky bit, and where I'm unfamiliar with any `perl` pecularities.  `ex` is specified by POSIX, and is part of the Single Unix Specification, so it's common and uniform.  Most importantly, I'm familiar with it, so it is what I went with.   Perl may well be a better option; if you are comfortable enough with it, please feel free to submit a separate PR for the `Makefile` that uses it instead.  I won't be offended, and the maintainer can choose what he/she deems best.

Ask `n` engineers to solve a problem, and you'll get `3*n` different solutions, on average.  :-)

Reply:
I removed the `ex` stuff and went back to the old `sed` solution because, at least currently, it's working for me on macOS _without_ GNU `sed` (i.e. _with_ BSD `sed`).